In [5]:
import numpy as np

def relu(x):
    return np.maximum(0, x)

# Données d'entrée
inputs = np.array([2000, 3])

# Paramètres du réseau
w1 = np.array([0.5, 0.7])
b1 = 10000
w2 = 0.6
b2 = 20000
w3 = 1.2
b3 = 30000

# Calculs
# Couche 1
z1 = np.dot(inputs, w1) + b1
a1 = relu(z1)
print("\nCouche 1:")
print(f"z1 = ({inputs[0]} × {w1[0]}) + ({inputs[1]} × {w1[1]}) + {b1}")
print(f"z1 = {z1}")
print(f"a1 (après ReLU) = {a1}")

# Couche 2
z2 = a1 * w2 + b2
a2 = relu(z2)
print("\nCouche 2:")
print(f"z2 = {a1} × {w2} + {b2}")
print(f"z2 = {z2}")
print(f"a2 (après ReLU) = {a2}")

# Couche de sortie
prediction = a2 * w3 + b3
print("\nCouche de sortie:")
print(f"prediction = {a2} × {w3} + {b3}")
print(f"Prédiction finale = {prediction:.2f} dollars")

# Résumé
print("\nRésumé de la propagation avant:")
print(f"1. Entrées: {inputs}")
print(f"2. Sortie couche 1 (après ReLU): {a1}")
print(f"3. Sortie couche 2 (après ReLU): {a2}")
print(f"4. Prédiction finale: {prediction:.2f} dollars")







Couche 1:
z1 = (2000 × 0.5) + (3 × 0.7) + 10000
z1 = 11002.1
a1 (après ReLU) = 11002.1

Couche 2:
z2 = 11002.1 × 0.6 + 20000
z2 = 26601.260000000002
a2 (après ReLU) = 26601.260000000002

Couche de sortie:
prediction = 26601.260000000002 × 1.2 + 30000
Prédiction finale = 61921.51 dollars

Résumé de la propagation avant:
1. Entrées: [2000    3]
2. Sortie couche 1 (après ReLU): 11002.1
3. Sortie couche 2 (après ReLU): 26601.260000000002
4. Prédiction finale: 61921.51 dollars


Analyse des résultats :

Première couche :
Entrées : 2000 m² et 3 chambres
Calcul : (2000 × 0.5) + (3 × 0.7) + 10000 = 11002.1
ReLU maintient cette valeur positive

Deuxième couche :
Entrée : 11002.1
Calcul : 11002.1 × 0.6 + 20000 = 26601.26
ReLU maintient cette valeur positive

Couche de sortie :
Entrée : 26601.26
Calcul final : 26601.26 × 1.2 + 30000 = 61921.51

La prédiction finale de 61,921.51 dollars représente le prix estimé de la maison basé sur :
- La surface (2000 m²)
- Le nombre de chambres (3)
- Les poids et biais appris par le réseau